### PR TITLE
feat(tabby-ui): Add a source code browser for ~/.tabby/repositories

### LINKED ISSUE #538: [TAB-438] Add a source code browser for ~/.tabby/repositories
**Please describe the feature you desire:**

To implement the desired feature, modify the `~/.tabby/config.toml` configuration file as shown below:

```toml
[[repositories]]
git_url = "https://github.com/TabbyML/tabby.git"
```

After executing `tabby scheduler`, it will generate a `~/.tabby/repositories` directory with the following content:

```bash
meng@Mengs-MacBook-Pro:~/.tabby/repositories % ls
https_github.com_TabbyML_tabby.git
```

#### with Monaco
Example application: [React Monaco File Tree](https://codesandbox.io/s/react-monaco-file-tree-ww9kis?file=/src/App.tsx)

- Backend is will implemented by #815 

- [HoverProvider](https://stackoverflow.com/questions/49449401/monaco-editor-hoverprovider-get-the-word-mouse-hovers-over) to provide intelligence display for tags.

- [DocumentSymbolProvider](https://microsoft.github.io/monaco-editor/typedoc/interfaces/languages.DocumentSymbolProvider.html) to provides symbol navigation functionality

#### with CodeMirror

Example application: https://github.com/vizhub-core/vzcode

- [Hover Tooltip](https://codemirror.net/examples/tooltip/)

<sub>[TAB-438](https://linear.app/tabbyml/issue/TAB-438/add-a-source-code-browser-for-tabbyrepositories)</sub>

### PR DESCRIPTION
**Pull request by liangfung**
Closes #538 

Considering the size of third-party editor and the degree of customization, I chose CodeMirror.

Entry: '/broswer'

Including:
1. File Tree
2. Code editor with tags

### CODE REVIEW

**Comment by wsxiaoys** [comment]
1. Support fronend routing: e.g /browser/tabby/README.md should bring me to the content of README.md
2. Seems to me the visual representation of current outline is limited, let's get back to more lightweight solution like hover tooltip: https://codemirror.net/examples/tooltip/ and maybe underline?

**Comment by liangfung** [comment]
> 1. Support fronend routing: e.g /browser/tabby/README.md should bring me to the content of README.md

implementation: /browser?repo=https_github.com_TabbyML_tabby.git&path=crates%2Faim-downloader%2Fsrc%2Faddress.rs

**Comment by wsxiaoys** [comment]
UI review locates at: https://www.figma.com/file/bxk2x2zKGppsyHyshT0Wzw/Source-Code-Browser?type=design&node-id=0-1&mode=design&t=jpVhFiMN8dN7sw0i-0

File: ee/tabby-ui/app/browser/page.tsx
Comment (wsxiaoys) [inline ee/tabby-ui/app/browser/page.tsx:1]: making this `/files/page.tsx`

File: ee/tabby-ui/components/header.tsx [hunk 1]
```diff
@@ -36,6 +36,7 @@ export function Header() {
           </span>
         </Link>
         <HeaderLink href="/api">API</HeaderLink>
+        <HeaderLink href="/browser">Browser</HeaderLink>
```
Comment (wsxiaoys) [inline ee/tabby-ui/components/header.tsx:39]:
```suggestion
        <HeaderLink href="/files">Files</HeaderLink>
```

File: ee/tabby-ui/app/browser/components/file-tree.tsx [hunk 2]
```diff
@@ -0,0 +1,568 @@
...
+        initDefaultEntries(data)
+      },
+      onError: () => {
+        initDefaultEntries()
+      },
+      revalidateOnMount: true
+    })
+
+  if (!initialized)
+    return <div className="mt-1 flex justify-center">loading...</div>
```
Comment (wsxiaoys) [inline ee/tabby-ui/app/browser/components/file-tree.tsx:446]: use https://ui.shadcn.com/docs/components/progress

**Pull request merged**